modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/consensus/ConsensusRumorHandlers.scala

```diff
@@ -0,0 +1,75 @@
+package org.tessellation.node.shared.infrastructure.consensus
+
+import cats.Show
+import cats.effect.Async
+import cats.syntax.all._
+
+import scala.reflect.runtime.universe.TypeTag
+
+import org.tessellation.node.shared.domain.consensus.ConsensusFunctions
+import org.tessellation.node.shared.infrastructure.consensus.declaration._
+import org.tessellation.node.shared.infrastructure.consensus.message._
+import org.tessellation.node.shared.infrastructure.gossip.RumorHandler
+import org.tessellation.security.SecurityProvider
+
+import io.circe.Decoder
+
+class ConsensusRumorHandlers[F[
+  _
+]: Async: SecurityProvider, Event: TypeTag: Decoder, Key: Show: TypeTag: Decoder, Artifact: TypeTag: Decoder, Context: Decoder, Status, Outcome, Kind: Decoder: TypeTag](
+  storage: ConsensusStorage[F, Event, Key, Artifact, Context, Status, Outcome, Kind],
+  manager: ConsensusManager[F, Key, Artifact, Context, Status, Outcome, Kind],
+  fns: ConsensusFunctions[F, Event, Key, Artifact, Context]
+) {
+
+  val eventHandler = RumorHandler.fromPeerRumorConsumer[F, ConsensusEvent[Event]]() { rumor =>
```

Comment:
Handlers should work exactly the same as they did before. One addition is a `binarySignatureHandler` that's being used for currency but not for global. This is kind of a collection of available handlers that can than be hand picked by the developer (given that the `ConsensusResources` still remain as a common resources class for both consensuses)